src/main/java/bio/terra/cli/command/resources/create/AiNotebook.java

```diff
@@ -58,80 +70,245 @@
       description = "The Compute Engine machine type of this instance.")
   private String machineType;
 
-  @CommandLine.Option(
-      names = "--vm-image-project",
-      defaultValue = "deeplearning-platform-release",
-      description = "The ID of the Google Cloud project that this VM image belongs to.")
-  private String vmImageProject;
-
-  @CommandLine.Option(
-      names = "--vm-image-family",
-      defaultValue = "r-latest-cpu-experimental",
-      description =
-          "Use this VM image family to find the image; the newest image in this family will be used.")
-  private String vmImageFamily;
-
   @CommandLine.Option(
       names = "--post-startup-script",
       description =
           "Path to a Bash script that automatically runs after a notebook instance fully boots up. "
               + "The path must be a URL or Cloud Storage path, e.g. 'gs://path-to-file/file-name'")
   private String postStartupScript;
 
-  // TODO(PF-747): Add parameters for all options supported by WSM API.
+  @CommandLine.Option(
+      names = "--metadata",
+      description =
+          "Custom metadata to apply to this instance.\nBy default sets some jupyterlab extensions "
+              + "(installed-extensions=jupyterlab_bigquery-latest.tar.gz,jupyterlab_gcsfilebrowser-latest.tar.gz,jupyterlab_gcpscheduler-latest.tar.gz) "
+              + "and the Terra workspace id (terra-workspace-id=[WORKSPACE_ID]).")
+  private Map<String, String> metadata;
+
+  @CommandLine.ArgGroup(exclusive = true, multiplicity = "0..1")
+  AiNotebook.VmOrContainerImage vmOrContainerImage;
+
+  static class VmOrContainerImage {
+    @CommandLine.ArgGroup(
+        exclusive = false,
+        multiplicity = "1",
+        heading =
+            "Definition of a custom Compute Engine virtual machine image for starting a "
+                + "notebook instance with the environment installed directly on the VM.\n"
+                + "If neither this nor --container-* are specified, default to \n"
+                + "'--vm-image-project="
+                + DEFAULT_VM_IMAGE_PROJECT
+                + " --vm-image-family="
+                + DEFAULT_VM_IMAGE_FAMILY
+                + " %n")
```

Comment:
nit: stray single quote?
This is really helpful.

Reply:
closed single quote